src/Parser/Core/Modules/CastEfficiency.js

```diff
@@ -101,16 +151,20 @@ class CastEfficiency extends Analyzer {
     const cooldown = ability.castEfficiency.disabled ? null : ability.cooldown;
     const cooldownMs = !cooldown ? null : cooldown * 1000;
     const cdInfo = this._getCooldownInfo(ability);
+    const timeSpentCasting = cooldown && ability.charges < 2 ? this._getTimeSpentCasting(ability) : 0;
+    const timeWaitingOnGCD = cooldown && ability.charges < 2 && ability.isOnGCD ? this._getTimeWaitingOnGCD(ability) : 0;
```

Comment:
Why are spells with multiple charges excluded? I reckon if Riptide had two charges you'd still want to add the GCD time when it first goes on cooldown.

Reply:
The GCD time is after a proc resets the cooldown, but because the proc happens on a cast you are always locked out of actually casting the spell again until the next GCD. For instance, take mangle and gore procs (other abilities has an X% chance to reset the CD o mangle):
![image](https://user-images.githubusercontent.com/6773230/41684953-54104ebe-74df-11e8-947a-a3b52fc04a4b.png)
The max possible casts is achieved here (as soon as another ability resets the CD it is cast the next GCD), but the time on cooldown is significantly lower.

If a spell has multiple charges, you could always have one recharging, even if the other one resets.